pressbooks/pressbooks
# Update sentry SDKs for PHP and JS

This change addresses: https://github.com/pressbooks/private/issues/437

NPM package for Sentry JS integration and PHP SDK were updated to the suggested versions through composer.

I also made a small change in the [sample rate parameter](https://docs.sentry.io/platforms/javascript/configuration/options/#sample-rate) to avoid float parsing issue in the JavaScript integration.

### How to test
- Set the environment variables for Sentry specified in: https://docs.google.com/document/d/1w3LYvcn0p-vBBNzLET-a_bDLX0RrJ9TKR-bQk4bYEh0/edit. Make sure you enabled JS and PHP monitoring.
- Add some syntax error in your code flow in PHP and [check in Sentry](https://sentry.io/organizations/pressbooks/issues/?environment=development&environment=staging&environment=production&project=3954026&statsPeriod=1h) to make sure those issues are displayed in the `development` environment.
- You also can generate JavaScript issues in the compiled js files (normally those are in `assets/dist/scripts/` folder). Make sure JavaScript issues also are [displayed in Sentry](https://sentry.io/organizations/pressbooks/issues/?environment=development&environment=staging&environment=production&project=3954026&statsPeriod=1h).
- Enable the environment variable for sample rate with the  max value `SENTRY_TRACE_SAMPLE_RATE=1` and check [the performance section](https://sentry.io/organizations/pressbooks/performance/?environment=development&environment=staging&environment=production&project=3954026&statsPeriod=1h) and make sure the transactions are tracked.
